### PR TITLE
`bug` command.

### PR DESCRIPTION
**Pull request by ldez**
### Description

In the `bug` command, some fields (like token, password, ...) are not anonymized.

This PR fixes this problem.

### CODE REVIEW

File: provider/mesos/mesos.go [hunk 1]
```diff
@@ -32,12 +32,13 @@ type Provider struct {
...
-	GroupsAsSubDomains bool   `description:"Convert Mesos groups to subdomains"`
-	ZkDetectionTimeout int    `description:"Zookeeper timeout (in seconds)"`
-	RefreshSeconds     int    `description:"Polling interval (in seconds)"`
-	IPSources          string `description:"IPSources (e.g. host, docker, mesos, rkt)"` // e.g. "host", "docker", "mesos", "rkt"
-	StateTimeoutSecond int    `description:"HTTP Timeout (in seconds)"`
+	ExposedByDefault   bool   `description:"Expose Mesos apps by default" export:"true"`
+	GroupsAsSubDomains bool   `description:"Convert Mesos groups to subdomains" export:"true"`
+	ZkDetectionTimeout int    `description:"Zookeeper timeout (in seconds)" export:"true"`
+	RefreshSeconds     int    `description:"Polling interval (in seconds)" export:"true"`
+	// e.g. "host", "docker", "mesos", "rkt"
```
Comment (mmatur) [inline provider/mesos/mesos.go]: Unnecessary comment, already present in IPSources description

File: types/types.go [hunk 2]
```diff
@@ -198,22 +198,22 @@ func NewConstraint(exp string) (*Constraint, error) {
...
+			return nil, errors.New("constraint must be tag-based. Syntax: tag==us-*")
 		}
 
 		constraint.Key = kv[0]
 		constraint.Regex = kv[1]
 		return constraint, nil
 	}
 
-	return nil, errors.New("Incorrect constraint expression: " + exp)
+	return nil, fmt.Errorf("incorrect constraint expression: %s", exp)
```
Comment (mmatur) [inline types/types.go]: To me errors should be homogeneous in all the file.

File: types/types.go [hunk 3]
```diff
@@ -198,22 +198,22 @@ func NewConstraint(exp string) (*Constraint, error) {
...
-		return nil, errors.New("Constraint expression missing valid operator: '==' or '!='")
+		return nil, errors.New("constraint expression missing valid operator: '==' or '!='")
 	}
 
 	kv := strings.SplitN(exp, sep, 2)
 	if len(kv) == 2 {
 		// At the moment, it only supports tags
 		if kv[0] != "tag" {
-			return nil, errors.New("Constraint must be tag-based. Syntax: tag==us-*")
+			return nil, errors.New("constraint must be tag-based. Syntax: tag==us-*")
```
Comment (mmatur) [inline types/types.go]: To me errors should be homogeneous in all the file.

File: types/types.go [hunk 4]
```diff
@@ -198,22 +198,22 @@ func NewConstraint(exp string) (*Constraint, error) {
 		sep = "!="
 		constraint.MustMatch = false
 	} else {
-		return nil, errors.New("Constraint expression missing valid operator: '==' or '!='")
+		return nil, errors.New("constraint expression missing valid operator: '==' or '!='")
```
Comment (mmatur) [inline types/types.go]: To me errors should be homogeneous in all the file.

File: types/types.go [hunk 5]
```diff
@@ -258,7 +258,7 @@ func (c *Constraint) MatchConstraintWithAtLeastOneTag(tags []string) bool {
 func (cs *Constraints) Set(str string) error {
 	exps := strings.Split(str, ",")
 	if len(exps) == 0 {
-		return errors.New("Bad Constraint format: " + str)
+		return fmt.Errorf("bad Constraint format: %s", str)
```
Comment (mmatur) [inline types/types.go]: To me errors should be homogeneous in all the file.

**Comment by ldez** [comment]
@emilevauge done.

File: cmd/traefik/anonymize/anonymize.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+// isExported return true is the field (from fieldName) is exported, else false
+func isExported(fieldName string) bool {
+	if len(fieldName) < 1 {
+		return false
+	}
+	return string(fieldName[0]) == strings.ToUpper(string(fieldName[0]))
+}
+
+func marshal(obfuscatedConfig interface{}, indent bool) ([]byte, error) {
```
Comment (mmatur) [inline cmd/traefik/anonymize/anonymize.go]: Should be anomConfig no ?

**Comment by mmatur** [comment]
:clap: LGTM

**Pull request merged**